File: uRides/uRides/cardViewController.m
//
//  cardViewController.m
//  uRides
//
//  Created by Colin Targonski on 3/14/16.
//  Copyright © 2016 uRides Corporation. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "cardViewController.h"
#import "CardIO.h"

@interface cardViewController : UIViewController<CardIOPaymentViewControllerDelegate>

// SomeViewController.m

- (void)viewWillAppear:(BOOL) {
    [super viewWillAppear:animated];
    [CardIOUtilities preload];
}


- (IBAction)scanCard:(id)sender {
    CardIOPaymentViewController *scanViewController = [[CardIOPaymentViewController alloc] initWithPaymentDelegate:self];
    [self presentViewController:scanViewController animated:YES completion:nil];
}


- (void)userDidCancelPaymentViewController:(CardIOPaymentViewController *)scanViewController {
    NSLog(@"User canceled payment info");
    // Handle user cancellation here...
    [scanViewController dismissViewControllerAnimated:YES completion:nil];
}

- (void)userDidProvideCreditCardInfo:(CardIOCreditCardInfo *)info inPaymentViewController:(CardIOPaymentViewController *)scanViewController {
    // The full card number is available as info.cardNumber, but don't log that!
    NSLog(@"Received card info. Number: %@, expiry: %02i/%i, cvv: %@.", info.redactedCardNumber, info.expiryMonth, info.expiryYear, info.cvv);
    // Use the card info...
    [scanViewController dismissViewControllerAnimated:YES completion:nil];
}

@end

File: uRides/uRides/cardViewController.swift
//
//  cardViewController.swift
//  uRides
//
//  Created by Colin Targonski on 3/14/16.
//  Copyright © 2016 uRides Corporation. All rights reserved.
//

import Foundation
import UIKit

class cardIOView: UIViewController, CardIOPaymentViewControllerDelegate {
    
    
    @IBOutlet weak var resultLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        CardIOUtilities.preload()
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func scanCard(sender: AnyObject) {
        let cardIOVC = CardIOPaymentViewController(paymentDelegate: self)
        cardIOVC.modalPresentationStyle = .FormSheet
        presentViewController(cardIOVC, animated: true, completion: nil)
    }
    
    func userDidCancelPaymentViewController(paymentViewController:
        CardIOPaymentViewController!) {
            resultLabel.text = "user canceled"
        paymentViewController?.dismissViewControllerAnimated(true, completion: nil)
    }
    
    // 3/14/16 as of now cardInfo is not being saved and is being passed as NULL, need cardInfo to create
    // token to stripe!!!!!!! -CT
    func userDidProvideCreditCardInfo(cardInfo: CardIOCreditCardInfo!, inPaymentViewController paymentViewController: CardIOPaymentViewController!) {
        
            let str = NSString(format: "Received card info.\n Number: %@\n expiry: %02lu/%lu\n cvv: %@.", cardInfo.redactedCardNumber, cardInfo.expiryMonth, cardInfo.expiryYear, cardInfo.cvv)
            resultLabel.text = str as String
        
        paymentViewController?.dismissViewControllerAnimated(true, completion: nil)
    }  
}